src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -218,9 +218,11 @@ public void InitializeHeaders()
         {
             if (FrameRequestHeaders == null)
             {
-                RequestHeaders = FrameRequestHeaders = new FrameRequestHeaders();
+                FrameRequestHeaders = new FrameRequestHeaders();
             }
 
+            RequestHeaders = FrameRequestHeaders;
```

Comment:
Question; why is this change only for request headers? Response headers are only reset like this [here](https://github.com/aspnet/KestrelHttpServer/blob/8df6fbc5009a6963864f042d552f3c4fdea4f2d7/src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs#L652), which means it'll only happen `if (_requestRejected || _applicationException != null && !HasResponseStarted)`. Am I not seeing something?